osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -234,18 +327,39 @@ private void applyOrigin(Anchor anchor)
         private Quad getSelectionQuad() =>
             GetSurroundingQuad(SelectedBlueprints.SelectMany(b => b.Item.ScreenSpaceDrawQuad.GetVertices().ToArray()));
 
-        private void applyAnchor(Anchor anchor)
+        private void applyAnchor(int hash)
         {
             foreach (var item in SelectedItems)
             {
                 var drawable = (Drawable)item;
 
-                var previousAnchor = drawable.AnchorPosition;
-                drawable.Anchor = anchor;
-                drawable.Position -= drawable.AnchorPosition - previousAnchor;
+                var anchor = getAnchorFromHashAndDrawableAndRecordWhetherUsingClosestAnchor(hash, drawable);
+
+                updateDrawableAnchor(drawable, anchor);
             }
         }
 
+        private static void updateDrawableAnchor(Drawable drawable, Anchor anchor)
+        {
+            var previousAnchor = drawable.AnchorPosition;
+            drawable.Anchor = anchor;
+            drawable.Position -= drawable.AnchorPosition - previousAnchor;
+        }
+
+        private Anchor getAnchorFromHashAndDrawableAndRecordWhetherUsingClosestAnchor(int hash, Drawable drawable)
```

Comment:
This method name is too much. Something like `mapHashToAnchorSettings` tells pretty much the same story. For expanded commentary, comments/xmldoc can be used, if necessary.